File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        return findMe(nums, target, 0, nums.length - 1);
    }

    private int findMe(int[] nums, int target, int start, int end) {
        if (start > end) return -1;
        int middle = (start + end) / 2;
        int val = nums[middle];

        if (val == target) return middle;
        else if (val > target) return findMe(nums, target, start, middle - 1);
        else return findMe(nums, target, middle + 1, end);
    }
}

File: Data Structures & Algorithms/sort-an-array/submission-0.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        def partition(start, end):
            pivot = nums[(start + end) // 2]
            l, r = start - 1, end + 1
            while l < r:
                while True:
                    l += 1
                    if nums[l] >= pivot:
                        break
                while True:
                    r -= 1
                    if nums[r] <= pivot:
                        break
                if l >= r:
                    return r

                nums[l], nums[r] = nums[r], nums[l]
            return r

        def quickSort(start, end):
            if start >= end:
                return
            i = partition(start, end)
            quickSort(start, i)
            quickSort(i + 1, end)

        quickSort(0, len(nums) - 1)

        return nums